paho/client.go

```diff
@@ -897,15 +898,29 @@ func (c *Client) publishQoS12(ctx context.Context, pb *packets.Publish, o Publis
 	if _, err := pb.WriteTo(c.config.Conn); err != nil {
 		c.debug.Printf("failed to write packet %d to connection: %s", pb.PacketID, err)
 		if o.Method == PublishMethod_AsyncSend {
-			return nil, ErrNetworkErrorAfterStored // Async send, so we don't wait for the response (may add callbacks in the future to enable user to obtain status)
+			if o.OnComplete != nil {
+				go o.OnComplete(nil, fmt.Errorf("failed to write packet to connection: %w", err))
```

Comment:
The packet will be sent when the connection is reestablished (assuming the session is not cleared). That's really the point of returning this error - it means "I could not send the packet right now but it's in the session state". If the callback will be called here then I think the same error should be passed (or it should wrap `ErrNetworkErrorAfterStored`).